google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadOfflineConversion.java

```diff
@@ -165,8 +203,27 @@ private void runExample(
             .setConversionAction(conversionActionResourceName)
             .setConversionDateTime(conversionDateTime)
             .setConversionValue(conversionValue)
-            .setCurrencyCode("USD")
-            .setGclid(gclid);
+            .setCurrencyCode("USD");
+
+    // Verifies that exactly one of gclid, gbraid, and wbraid is specified, as required.
+    // See https://developers.google.com/google-ads/api/docs/conversions/upload-clicks for details.
+    long numberOfIdsSpecified =
+        Arrays.asList(gclid, gbraid, wbraid).stream().filter(idField -> idField != null).count();
+    if (numberOfIdsSpecified > 1) {
```

Comment:
What if none are set? Should this be if (numberOfIdsSpecified != 1) since at least one must be used?